api/envoy/service/auth/v2/BUILD

```diff
@@ -26,3 +26,11 @@ api_proto_library_internal(
         "//envoy/type:http_status",
     ],
 )
+
+api_cc_grpc_library(
```

Comment:
Why can't we build this under the hood of `api_proto_library_internal`, so we don't need to do explicit declaration for each of these? I'd prefer that things "just work" rather than having to boiler plate each individual thing we want to make a service consumable outside of Envoy.

Reply:
This is something I pondered. I opted for this route because I thought it was unlikely that most services would require CC protos and services. Is the preference to add to `api_proto_library` a `require_cc` option analogous to `require_py`?

Reply:
Yep, I think that would be a good idea (the additional option). Maybe call it `grpc_services` and make it a bool.

Reply:
@htuch and @lizan see latest changes whereby `api_cc_grpc_library` and `api_go_grpc_library` are used by `api_proto_library` when `has_services` is set. I've refactored some of the services so they make use of this change. Some services that are Go-specific have remained the same so they can specialize calls to `api_go_grpc_library` as necessary.